src/layerelement.cpp

```diff
@@ -2517,6 +2545,11 @@ int LayerElement::LayerElementsInTimeSpan(FunctorParams *functorParams) const
     // The element is starting after the event end - we can stop here
     if (time >= (params->m_time + params->m_duration)) return FUNCTOR_STOP;
 
+    if (this->Is(NOTE) && this->GetParent()->Is(CHORD)
```

Comment:
You should use `IsChordTone()` or look for the first ancestor (and not the parent)